assets/src/data/services/apollo/queries/datetimes/useDatetimesQuery.ts

```diff
@@ -0,0 +1,14 @@
+import { QueryHookOptions } from '@apollo/react-hooks';
+import { QueryResult } from '@apollo/react-common';
+
+import { useCacheQuery } from '@dataServices/apollo/queries';
+import { DatetimesList, DatetimesQueryArgs } from './types';
+import { EntityEdge } from '@dataServices/types';
+
+const useDatetimesQuery = <DatetimeEdge extends EntityEdge>(
+	queryOptions: QueryHookOptions<DatetimesList<DatetimeEdge>, DatetimesQueryArgs>
+): QueryResult<DatetimesList<DatetimeEdge>, DatetimesQueryArgs> => {
+	return useCacheQuery<DatetimesList<DatetimeEdge>>(queryOptions);
+};
```

Comment:
I've found an example on how some pieces can be extracted, not sure if the same can be done here:
```
type TypeWithGeneric<T> = T[]
type extractGeneric<Type> = Type extends Array<infer X> ? X : null
    
type extracted = extractGeneric<TypeWithGeneric<number>>
```
but, if possible, this needs to be adjusted a little

Reply:
Let me try that... Thanks

Reply:
Done